net/respondd/src/respondd.c

```diff
@@ -41,150 +49,395 @@
 #include <sys/socket.h>
 
 
+struct provider_list {
+	struct provider_list *next;
+
+	char *name;
+	respondd_provider provider;
+};
+
+struct request_type {
+	struct provider_list *providers;
+
+	struct json_object *cache;
+	uint64_t cache_time;
+	int64_t cache_timeout;
+};
+
+
+static int64_t now;
+static struct hsearch_data htab;
+
+
+static struct json_object * merge_json(struct json_object *a, struct json_object *b);
+
+
 static void usage() {
-  puts("Usage: respondd [-h] -g <group> -p <port> -i <if0> [-i <if1> ..]");
-  puts("  -g <ip6>         multicast group, e.g. ff02::2:1001");
-  puts("  -p <int>         port number to listen on");
-  puts("  -i <string>      interface on which the group is joined");
-  puts("  -c <string>      Lua command");
-  puts("  -h               this help\n");
+	puts("Usage:");
+	puts("  respondd -h");
+	puts("  respondd [ -p <port> ] [ -g <group> ] -i <if0> [-i <if1> ..] [ -d <dir> ]");
```

Comment:
Sorry, but this still isn't right. Neither is `-i` strictly required (if it is not given, respondd only replies to unicast and broadcast and doesn't join any multicast groups, which I consider a good thing), nor may `-i` be given without `-g` (would output "Multicast group must be given before interface." in line 409). While the other way round (`-g` without `-i`) wouldn't throw an error, it doesn't really make any sense. So I still propose the following:
```
Usage:
  respondd -h
  respondd [-p <port>] [-g <group> -i <if0> [-i <if1> ..]] [-d <dir>]
```